pkg/reconciler/route/domains/domains.go

```diff
@@ -54,13 +55,22 @@ func DomainNameFromTemplate(ctx context.Context, r *v1alpha1.Route, name string)
 	domainConfig := config.FromContext(ctx).Domain
 	domain := domainConfig.LookupDomainForLabels(r.ObjectMeta.Labels)
 	annotations := r.ObjectMeta.Annotations
+
+	// Generates an unique string that can be used as part of domain name
+	// The unique string is SHA256 generated from KnRoute's UID and shortened to 5 symbols
+	var unique string
+	if r.UID != "" {
```

Comment:
just wondering... would UID ever be blank?

Reply:
I've added it as a precation for the original `base64.RawURLEncoding` solution, but I can remove, since sha256 generates hash even from the blank string.